src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageAssets.cs

```diff
@@ -1405,6 +1428,121 @@ private void WritePackageDependencies()
+
+                    if (string.IsNullOrEmpty(package.Name) || !allowItemSpecs.Contains(packageId))
+                    {
+                        // Name is required
+                        // We are not interested in this definition (not top-level, or wrong target)
+                        continue;
+                    }
+
+                    var dependencyType = GetDependencyType(package.Type);
+
+                    if (dependencyType == DependencyType.Package ||
+                        dependencyType == DependencyType.Unresolved)
+                    {
+                        WriteItem(packageId);
+                        WriteMetadata(MetadataKeys.Name, package.Name);
+
+                        var version = packageVersion ?? string.Empty;
+                        WriteMetadata(MetadataKeys.Version, version);
+
+                        var isImplicitlyDefined = implicitPackageReferences.Contains(package.Name);
+                        WriteMetadata(MetadataKeys.IsImplicitlyDefined, isImplicitlyDefined.ToString());
+
+                        string resolvedPackagePath = _packageResolver.GetPackageDirectory(package.Name, package.Version);
+                        var resolvedPath = resolvedPackagePath ?? string.Empty;
+                        var resolved = !string.IsNullOrEmpty(resolvedPath);
+                        WriteMetadata(MetadataKeys.Resolved, resolved.ToString());
+
+                        string itemPath = package.Path ?? string.Empty;
+                        var path = (resolved
+                            ? resolvedPath
+                            : itemPath) ?? string.Empty;
+                        WriteMetadata(MetadataKeys.Path, path);
+
+                        string itemDiagnosticLevel = GetPackageDiagnosticLevel(package);
+                        var diagnosticLevel = itemDiagnosticLevel ?? string.Empty;
+                        WriteMetadata(MetadataKeys.DiagnosticLevel, diagnosticLevel);
+                    }
+                }
+
+                List<string> GetPackageDependencies()
+                {
+                    List<string> dependencies = new();
+                    HashSet<string> projectFileDependencies = _lockFile.GetProjectFileDependencySet(_compileTimeTarget.Name);
+
+                    var resolvedPackageVersions = _compileTimeTarget.Libraries
+                        .ToDictionary(pkg => pkg.Name, pkg => pkg.Version.ToNormalizedString(), StringComparer.OrdinalIgnoreCase);
+
+                    string frameworkAlias = _lockFile.GetLockFileTargetAlias(_compileTimeTarget);
+
+                    foreach (var package in _compileTimeTarget.Libraries)
+                    {
+                        if (projectFileDependencies.Contains(package.Name))
+                        {
+                            string packageId = GetPackageId(package);
+
+                            // itemSpec
+                            dependencies.Add(packageId);
```

Comment:
nit: Use the variable name to document the value's semantic meaning, instead of a separate comment.

```suggestion
                            string itemSpec = GetPackageId(package);

                            dependencies.Add(itemSpec);
```